### PR TITLE
Getting Started with Image Preprocessing in python

### LINKED ISSUE #3188: Getting Started with Image Data Pre-processing in Python
### Proposed title of article
Getting Started with Image Data Pre-processing in Python

### Introduction paragraph (2-3 paragraphs):
In a machine learning project, we always start with data collection followed by data preprocessing or cleaning step
which needs to be gotten right for a better performing model. There are several techniques used to preprocess data (in this case image data). Examples include; image resizing, converting images to grayscale, etc. 

In this tutorial, we shall be looking at data pre-processing of images. Image preprocessing involves converting image data into a form that allows machine learning algorithms to solve it. It is often used to reduce the complexity and increase the accuracy of the applied algorithm.  

### Key takeaways: 
*  Understanding image preprocessing techniques and processes 
*  Data standardization
*  Geometric and color transformation or converting color to grayscale
*  Data augmentation

### References: 
N/A

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by FREDERICO23**
### Article checklist

**ATTENTION: In your PR - add the ISSUE (topic form) # that this PR corresponds to. You can do this by also adding a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.**
      
      
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**If you have not already please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve our overall technical writing.**

#### Preliminary Checks - Formatting and Structure 
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**

This fixes #3188

### CODE REVIEW

**Comment by lalith1403** [comment]
Hello @FREDERICO23. There are several grammatical errors and many paragraphs that are tough to comprehend. Kindly check and change them. Once done, ping me.

**Comment by hectorkambow** [comment]
plagiarism check done

**Comment by FREDERICO23** [comment]
@lalith1403 Revisions done

File: content/articles/image-preprocessing-in-python/index.md [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+gray_image = skimage.color.rgb2gray(image)
+plt.imshow(gray_image, cmap = 'gray')
+```
+
+Output
+![gray_scale](/engineering-education/image-preprocessing-in-python/gray_scale.jpg)
+
+>**NB**: Converting images to grayscale might not always be practical in solving some problems. A few examples where it would impractical to use grayscale include: _traffic lights_, _healthcare diagnosis_, _autonomous vehicles_, _agriculture_, etc. The best way to know whether to use it or not depends on your human visual ability to be able to identify an object without colour.
+
+#### Normalization
```
Comment (lalith1403) [inline content/articles/image-preprocessing-in-python/index.md:146]: Explain why normalization works! Why would rescaling [0,255] to [-1,1] or [0,1] help? How would it help? Do this for all the techniques mentioned.

**Review by lalith1403 — CHANGES_REQUESTED** [review]
Very well-written article. Explain why the techniques work. How would data augmentation help? Why would random transformation to images help?

**Comment by FREDERICO23** [comment]
Changes Done

**Comment by lalith1403** [comment]
Hello @FREDERICO23. I had made a lot of changes in the earlier commit. You have updated the changes without pulling my changes. Please can you consider pulling the changes made in my commit and then push your changes.

**Comment by hectorkambow** [comment]
@lalith1403 
Can we get a status on this PR

**Comment by WanjaMIKE** [comment]
Nice article @FREDERICO23. Good review @lalith1403

**Comment by WanjaMIKE** [comment]
Merged #3306

**Pull request merged**